addons/link_tracker/controller/main.py

```diff
@@ -5,15 +5,23 @@
 from odoo.http import request
 
 
+ignore_user_agent_prefix = ['twitterbot', 'facebookexternalhit', 'facebot',
+                            'linkedinbot', 'whatsapp', 'googlebot', 'pinterest', 'instagram']
+
+
 class LinkTracker(http.Controller):
 
     @http.route('/r/<string:code>', type='http', auth='public', website=True)
     def full_url_redirect(self, code, **post):
-        country_code = request.session.geoip and request.session.geoip.get('country_code') or False
-        request.env['link.tracker.click'].sudo().add_click(
-            code,
-            ip=request.httprequest.remote_addr,
-            country_code=country_code
-        )
+
+        user_agent = str(request.httprequest.user_agent) or ""
```

Comment:
```suggestion
        user_agent = str(request.httprequest.user_agent) or ''
```

Reply:
why do you need to call `str` ?

Reply:
because `request.httprequest.user_agent` is of class `<class 'werkzeug.useragents.UserAgent'>`, and we need to call .lower() on that.

Reply:
I changed it to `request.httprequest.user_agent.string`

Reply:
so you do not need `or ''` anymore right ?

Reply:
Yep that works ! I was afraid that if no User Agent was sent it would be None and crash when comparing to a string, but turns out it's just an empty string when not set